egs/wsj/s5/steps/cleanup/internal/get_pron_stats.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+
+# Copyright 2016  Xiaohui Zhang
+# Apache 2.0.
+
+from __future__ import print_function
+import argparse
+import sys
+import warnings
+
+# Collect pronounciation stats from a ctm_prons.txt file of the form output
+# by steps/cleanup/debug_lexicon.sh.  This input file has lines of the form:
+#  utt_id word phone1 phone2 .. phoneN
+#  e.g.
+#  foo-bar123-342  hello h eh l l ow
+# (and this script does require that lines from the same utterance be ordered in
+# order of time).
+# The output of this program is word pronunciation stats of the form:
+#  count word phone1 .. phoneN
+#  e.g.:
+#  24.0  hello h ax l l ow
+# This program uses various heuristics to account for the fact that in the input ctm_prons.txt
+# file may not always be well aligned.  As a result of some of these heuristics the counts will
+# not always be integers.
+
+def GetArgs():
+    parser = argparse.ArgumentParser(description = "Accumulate pronounciation statistics from "
+                                     "a ctm_prons.txt file.",
+                                     epilog = "See steps/cleanup/debug_lexicon.sh for example")
+    parser.add_argument("ctm_prons_file", metavar = "<ctm-prons-file>", type = str,
+                        help = "File containing word-pronounciation alignments obtained from a ctm file; "
+                        "each line must be <utt_id> <word> <phones>")
+    parser.add_argument("silence_file", metavar = "<silphone-file>", type = str,
+                        help = "File containing a list of silence phones.")
+    parser.add_argument("optional_silence_file", metavar = "<optional_silence>", type = str,
+                        help = "File containing the optional silence phone. We'll be replacing empty prons by this.")
```

Comment:
(from our old thread): 
> Sorry I forgot to mention, this is the candidate generating phase so we have
> to remove empty prons or merge them with SIL. So do you think we should
> merge them at this point? Actually I think simply removing empty prons may
> make sense also, though the file counts.txt generated later won't reflect
> the real counts.

Oh.  Yes, empty prons would cause a problem for lattice word
alignment.  So maybe turn them into SIL as you suggested.  I think we
should keep all the counts.